### PR TITLE
EUI-5246 Create mock API for post hearing actuals

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-5246

### Change description ###
Create mock API for post hearing actuals

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/hearings/services/hearings.service.ts [hunk 1]
```diff
@@ -52,4 +52,8 @@ export class HearingsService {
   public updateHearingActuals(hearingId: string, hearingActualsModel: HearingActualsModel): Observable<HearingActualsModel> {
     return this.http.put<HearingActualsModel>(`api/hearings/hearingActuals/${hearingId}`, hearingActualsModel);
   }
+
+  public submitHearingActuals(hearingId: string): Observable<any> {
```
Comment (rayliangatos) [inline src/hearings/services/hearings.service.ts:56]: The any needs to be replaced with the actual model.

File: api/hearings/index.ts [hunk 2]
```diff
@@ -126,3 +126,18 @@ export async function updateHearingActuals(req: EnhancedRequest, res: Response,
...
+
+/**
+ * submitHearingActuals - submit hearing actuals
+ */
+export async function submitHearingActuals(req: EnhancedRequest, res: Response, next: NextFunction) {
+  const reqBody = req.body;
+  const hearingId = req.query.hearingId;
+  const markupPath = `${hearingsUrl}/hearingActualsCompletion/${hearingId}`;
+  try {
+    const { status, data }: { status: number, data: any } = await handlePost(markupPath, reqBody, req);
```
Comment (rayliangatos) [inline api/hearings/index.ts:138]: data: any if that can be a model? could you check the LLD API contract?

File: src/hearings/services/hearings.service.ts [hunk 3]
```diff
@@ -52,4 +52,8 @@ export class HearingsService {
...
+    return this.http.post<any>(`api/hearings/hearingActualsCompletion/${hearingId}`, {});
```
Comment (rayliangatos) [inline src/hearings/services/hearings.service.ts:57]: as chat body will be null as the Back end api do not accept any body including {}.

File: api/hearings/index.ts [hunk 4]
```diff
@@ -126,3 +126,18 @@ export async function updateHearingActuals(req: EnhancedRequest, res: Response,
     next(error);
   }
 }
+
+/**
+ * submitHearingActuals - submit hearing actuals
+ */
+export async function submitHearingActuals(req: EnhancedRequest, res: Response, next: NextFunction) {
+  const reqBody = req.body;
```
Comment (rayliangatos) [inline api/hearings/index.ts:134]: reqBody couldn't set as null as per API contract.

File: src/hearings/services/hearings.service.ts (hunk 3 above)
Comment (johnbenjamin-hmcts) [reply src/hearings/services/hearings.service.ts]: @rayliangatos - as discussed I have updated the code to use null.

File: api/hearings/index.ts (hunk 4 above)
Comment (johnbenjamin-hmcts) [reply api/hearings/index.ts]: @rayliangatos - as discussed I have updated the code to use null.

File: src/hearings/services/hearings.service.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply src/hearings/services/hearings.service.ts]: I have updated the code to use number instead of any, as the API returns only the status.

Comment (johnbenjamin-hmcts) [reply src/hearings/services/hearings.service.ts]: I have updated the code to use number instead of any, as the API returns only the status.

**Review by rayliangatos — APPROVED** [review]

**Review by Kartheek-S — APPROVED** [review]

**Pull request merged**